api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -1185,6 +1185,10 @@ message Cluster {
   // If ``connection_pool_per_downstream_connection`` is true, the cluster will use a separate
   // connection pool for every downstream connection
   bool connection_pool_per_downstream_connection = 51;
+
+  // If ``connection_pool_per_key_format`` is set, the cluster will use a separate connection
+  // pool for every unique string format
```

Comment:
Sorry, but I cannot get it from this comment. When will user need to set this field? And which scenario will need this field?

Reply:
The feature idea is to let user define a unique key string ( computed from request attributes & dynamic metadata ) to create an upstream connection pool. Ex: connection_pool_per_key_format: "%REQ(:host) + DYNAMIC_METADATA(test:key)%".

Reply:
Using a formatted string seems to me to be fragile (different patterns may end up with the same output string).
Wouldn't it be better to have a structure that defines the relevant fields that need to be used as a key?

Reply:
@adisuissa The thought is user will aware each pattern results in a unique string.  Open for defining a structure as well, can share an example or format for it ? 

something like below:
connection_pool_key_format:
  - key1: value
  - key2: value 
  
Computed unique string: "key1=value:key2=value"